pkg/controller/hosted/controller.go

```diff
@@ -102,6 +102,9 @@ func (r *ReconcileHosted) Reconcile(ctx context.Context, request reconcile.Reque
 	}
 
 	if !managedCluster.DeletionTimestamp.IsZero() {
+		if err := helpers.DeleteManagedClusterAddons(ctx, r.clientHolder.RuntimeClient, managedClusterName); err != nil {
```

Comment:
I think we should still go head deleting works even we have error with addon deletion. Maybe we should delete work and addon in one single func and return aggregated error.

Reply:
> I think we need import-controller to have permission to delete addons at first. Have we added that in installer?

@qiujian16 yes, delete rbac has been added into installer.